frontend/tests/unit/models/public-schools-analysis-test.js

```diff
@@ -0,0 +1,172 @@
+    assert.equal(analysis.indirectEffect, true) // if esEffect or hsEffect are true
+  });
+  //
+  test('calculates hsEffect correctly', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    assert.equal(project.netUnits, 450) // totalUnits - seniorUnits = 500 - 50 = 450
+    assert.equal(analysis.estHsStudents, 41) // math.ceil(currentMultiplier.hs * project.netUnits) = 0.09 * 450
+    assert.equal(analysis.hsEffect, false); // if multipliers.thresholdHsStudents > estHsStudents, then false
+    assert.equal(analysis.indirectEffect, true); // if esEffect or hsEffect are true
+  });
+  //
+  test('concatenates subdistricts from DB and subdistricts from user correctly', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis', 'subdistrictsFromUserAdded');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    assert.equal(analysis.subdistricts.length, 2)
+    assert.equal(analysis.subdistricts[1].id, 185);
+    assert.equal(analysis.multiSubdistrict, true);
+    assert.equal(analysis.subdistrictCartoIds[1], 24);
+    assert.equal(analysis.subdistrictSqlPairs[1], '(17, 5)') // `(${f.district}, ${f.subdistrict})`
+  });
+  //
+  test('concatenates buildings and allSchools correctly', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    const bluebookBuildings = analysis.buildings.filter(obj => { return obj.source === 'bluebook'});
+    const lcgmsBuildings = analysis.buildings.filter(obj => { return obj.source === 'lcgms'});
+    const scaProjectsBuildings = analysis.buildings.filter(obj => { return obj.source === 'scaProjects'});
+    const bluebookSchools = analysis.allSchools.filter(obj => { return obj.source === 'bluebook'});
+    const lcgmsSchools = analysis.allSchools.filter(obj => { return obj.source === 'lcgms'});
+
+    // buildings concatenates bluebook, lcgms, and scaProjects
+    assert.equal(bluebookBuildings[0].name, 'I.S. 2 - K')
+    assert.equal(lcgmsBuildings[0].name, 'Banana Bonanza')
+    assert.equal(scaProjectsBuildings[1].name, 'Donuts Delight')
+
+    // allSchools concatenates bluebook and lcgms
+    assert.equal(bluebookSchools[0].name, 'I.S. 2 - K')
+    assert.equal(lcgmsSchools[0].name, 'Banana Bonanza')
```

Comment:
while I wouldn't say we need to explicitly test that any external function "works", if we are using a function to achieve some behavior in our code (in this case, that null values are excluded from some variable?) then we should have a test that confirms that. The idea being if someone were to rewrite the code and accidentally leave out excluding of null values, then we want a test that will fail and catch that regression.